shared/sliceutil/slice_test.go

```diff
@@ -337,8 +342,12 @@ func TestIntersectionByteSlices(t *testing.T) {
 					{4, 5},
 					{4, 5},
 				},
+				{
+					{4, 5},
+					{4, 5},
+				},
 			},
-			result: [][]byte{{4, 5}},
+			result: [][]byte{{4, 5}, {4, 5}},
```

Comment:
> Terence, issue wasn't in tests but in the fact that intersection functions were modifying arguments (which was unexpected).
> 
> I've created alternative implementation #6067
> 
> We can, either:
> 
> * merge that alternative PR (and close this one unmerged)
> * fix your implementation by introducing temp var, and not modifying passed in arguments
> * if you feel strongly about validity of your implementation, I can update fetcher tests too

You are right. I should not have modified the first argument instead of just occupying a new one. Let's close this and use #6067. Thanks!